apps/examples/testcase/le_tc/kernel/tc_libc_math.c

```diff
@@ -2100,6 +2105,78 @@ static void tc_libc_math_log2l(void)
 }
 
 /**
+ * @fn                   :tc_libc_math_log10
+ * @brief                :Returns the binary (base-10) logarithm of x
+ * @Scenario             :Returns the binary (base-10) logarithm of x
+ * API's covered         :log10
+ * Preconditions         :None
+ * Postconditions        :None
+ * @return               :void
+ */
+static void tc_libc_math_log10(void)
+{
+	const double in_val[] = { 1.0, INFINITY, NAN, -1.0, 10.0 };
+	const double sol_val[] = { ZERO, INFINITY, NAN, NAN, 1.0 };
+	double ret_val[SIZE(sol_val, double)];
+	int log10_idx;
+
+	for (log10_idx = 0; log10_idx < SIZE(in_val, double); log10_idx++) {
+		ret_val[log10_idx] = log10(in_val[log10_idx]);
+		TC_ASSERT_LEQ("log10", fabs(sol_val[log10_idx] - ret_val[log10_idx]), FLT_EPSILON);
+	}
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @fn                   :tc_libc_math_log10f
+ * @brief                :Returns the binary (base-10) logarithm of x
+ * @Scenario             :Returns the binary (base-10) logarithm of x
+ * API's covered         :log10f
+ * Preconditions         :None
+ * Postconditions        :None
+ * @return               :void
+ */
+static void tc_libc_math_log10f(void)
+{
+	const float in_val[] = { 1.0, INFINITY, NAN, -1.0, 10.0 };
+	const float sol_val[] = { ZERO, INFINITY, NAN, NAN, 1.0 };
+	float ret_val[SIZE(sol_val, float)];
+	int log10f_idx;
+
+	for (log10f_idx = 0; log10f_idx < SIZE(in_val, float); log10f_idx++) {
+		ret_val[log10f_idx] = log10f(in_val[log10f_idx]);
+		TC_ASSERT_LEQ("log10f", fabsf(sol_val[log10f_idx] - ret_val[log10f_idx]), FLT_EPSILON);
+	}
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @fn                   :tc_libc_math_log10l
+ * @brief                :Returns the binary (base-10) logarithm of x
+ * @Scenario             :Returns the binary (base-10) logarithm of x
+ * API's covered         :log10l
+ * Preconditions         :None
+ * Postconditions        :None
+ * @return               :void
+ */
+static void tc_libc_math_log10l(void)
+{
+	const long double in_val[] = { 1.0, INFINITY, NAN, -1.0, 10.0 };
+	const long double sol_val[] = { ZERO, INFINITY, NAN, NAN, 1.0 };
+	long double ret_val[SIZE(sol_val, long double)];
+	int log10l_idx;
+
+	for (log10l_idx = 0; log10l_idx < SIZE(in_val, long double); log10l_idx++) {
```

Comment:
The ret_val is defined with sol_val size, but here is in_val. Is it right?

Reply:
As both in_val and sol_val size are same, I used it.
actually same changes are there in few other tc also. I will update them also.